onnxruntime/core/graph/contrib_ops/contrib_defs.cc

```diff
@@ -24,6 +24,17 @@
 #pragma warning(disable : 26426)
 #endif
 namespace ONNX_NAMESPACE {
+
+inline int64_t HandleNegativeAxis(int64_t axis, int64_t rank) {
+  if (axis >= rank || axis < -rank) {
+    fail_shape_inference("axis ", axis,
+                         " is not in valid range [-", rank, ",", rank - 1, "]");
```

Comment:
So the range is not symmetric?

Reply:
Is rank non-negative? Was it checked anywhere?

Reply:
the range is [-rank, rank-1]

Reply:
rank takes in the size of dim.